L1Trigger/L1TTrackMatch/plugins/L1TruthTrackFastJetProducer.cc

```diff
@@ -0,0 +1,218 @@
+  edm::ESGetToken<TrackerTopology, TrackerTopologyRcd> tTopoToken_;
+  const edm::EDGetTokenT<TTTrackAssociationMap<Ref_Phase2TrackerDigi_> > ttTrackMCTruthToken_;
+};
+
+// constructor
+L1TruthTrackFastJetProducer::L1TruthTrackFastJetProducer(const edm::ParameterSet& iConfig)
+    : trkZMax_((float)iConfig.getParameter<double>("trk_zMax")),
+      trkPtMin_((float)iConfig.getParameter<double>("trk_ptMin")),
+      trkEtaMax_((float)iConfig.getParameter<double>("trk_etaMax")),
+      trkNStubMin_((int)iConfig.getParameter<int>("trk_nStubMin")),
+      trkNPSStubMin_((int)iConfig.getParameter<int>("trk_nPSStubMin")),
+      coneSize_((float)iConfig.getParameter<double>("coneSize")),
+      displaced_(iConfig.getParameter<bool>("displaced")),
+      trackToken_(consumes<std::vector<TTTrack<Ref_Phase2TrackerDigi_> > >(
+          iConfig.getParameter<edm::InputTag>("L1TrackInputTag"))),
+      tTopoToken_(esConsumes<TrackerTopology, TrackerTopologyRcd>(edm::ESInputTag("", ""))),
+      ttTrackMCTruthToken_(consumes<TTTrackAssociationMap<Ref_Phase2TrackerDigi_> >(
+          iConfig.getParameter<edm::InputTag>("MCTruthTrackInputTag"))) {
+  if (displaced_)
+    produces<TkJetCollection>("L1TruthTrackFastJetsExtended");
+  else
+    produces<TkJetCollection>("L1TruthTrackFastJets");
+}
+
+// destructor
+L1TruthTrackFastJetProducer::~L1TruthTrackFastJetProducer() {}
+
+// producer
+void L1TruthTrackFastJetProducer::produce(edm::Event& iEvent, const edm::EventSetup& iSetup) {
+  std::unique_ptr<TkJetCollection> L1TrackFastJets(new TkJetCollection);
+
+  // L1 tracks
+  edm::Handle<std::vector<TTTrack<Ref_Phase2TrackerDigi_> > > TTTrackHandle;
+  iEvent.getByToken(trackToken_, TTTrackHandle);
+  std::vector<TTTrack<Ref_Phase2TrackerDigi_> >::const_iterator iterL1Track;
+
+  // MC truth
+  edm::Handle<TTTrackAssociationMap<Ref_Phase2TrackerDigi_> > MCTruthTTTrackHandle;
+  iEvent.getByToken(ttTrackMCTruthToken_, MCTruthTTTrackHandle);
+
+  // Tracker Topology
+  const TrackerTopology& tTopo = iSetup.getData(tTopoToken_);
+
+  fastjet::JetDefinition jet_def(fastjet::antikt_algorithm, coneSize_);
+  std::vector<fastjet::PseudoJet> JetInputs;
+
+  unsigned int this_l1track = 0;
+  for (iterL1Track = TTTrackHandle->begin(); iterL1Track != TTTrackHandle->end(); iterL1Track++) {
+    edm::Ptr<TTTrack<Ref_Phase2TrackerDigi_> > l1track_ptr(TTTrackHandle, this_l1track);
+    this_l1track++;
+    float trk_pt = iterL1Track->momentum().perp();
+    float trk_z0 = iterL1Track->z0();
+    std::vector<edm::Ref<edmNew::DetSetVector<TTStub<Ref_Phase2TrackerDigi_> >, TTStub<Ref_Phase2TrackerDigi_> > >
+        theStubs = iterL1Track->getStubRefs();
+    int trk_nstub = (int)theStubs.size();
+
+    // standard quality cuts
+    if (std::abs(trk_z0) > trkZMax_)
+      continue;
+    if (std::abs(iterL1Track->momentum().eta()) > trkEtaMax_)
+      continue;
+    if (trk_pt < trkPtMin_)
+      continue;
+    if (trk_nstub < trkNStubMin_)
+      continue;
```

Comment:
This is super picky, so I'm not going to insist on it, but it may be a bit clearer to put these conditions after each relevant variable above to clarify just a bit, and possibly save yourself microscopic amounts of computing on assigning variables if the first fails a condition anyways.

There is another block like this in other modules that may be a little more worthwhile on this, considering it may skip some loops.

Reply:
Done.